src/Behaviors/SourcesBehavior.php

```diff
@@ -21,6 +21,30 @@
  */
 class SourcesBehavior extends Behavior
 {
+    /**
+     * Recursively find sources in definition attributes.
+     *
+     * @param string $fieldType
+     * @param array  $attributes
+     * @param string $indexFrom
+     * @param string $indexTo
+     *
+     * @return array
+     */
+    public function findSources(string $fieldType, array $attributes, string $indexFrom, string $indexTo): array
+    {
+        foreach ($attributes as $key => $attribute) {
+            if ($key === 'source' || $key === 'sources') {
+                $method = 'get'.ucfirst($key);
```

Comment:
Is this really that much shorter than
```php
if ($key === 'source') {
    $attributes[$key] = $this->getSource($fieldType, $attribute, $indexFrom, $indexTo)
} elseif ($key === 'sources') {
    $attributes[$key] = $this->getSources($fieldType, $attribute, $indexFrom, $indexTo)
} elseif (is_array($attribute)) {
    $attributes[$key] = $this->findSources($fieldType, $attribute, $indexFrom, $indexTo);
}
```
The more exmplicit variant might be more readable